app/controllers/admin/spaces_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Admin::SpacesController < AdminAreaController
+
+  layout 'admin_area'
+
+  def index
+  end
+
+  def new
+    @space = Space.new
+  end
+
+  def create
+    space = Space.new(space_params)
+    if space.save
+      flash[:notice] = "Space created successfully!"
+    else
+      flash[:alert] = "Something went wrong."
+    end
+    redirect_to :back
+  end
+
+  def edit
```

Comment:
Isn't there an id param so you can not use :name and instead use :id ? I think it has to be since it will have been part of the path (eg: `/admin/spaces/123`)